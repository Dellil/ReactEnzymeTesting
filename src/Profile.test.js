import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Profile username="aksaksdm" name="장찬희" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders username and name', () => {
        const wrapper = mount(<Profile username="velopert" name="ㅈㅊㅎ"/>);
        // react 컴포넌트의 prop 검사
        expect(wrapper.props().username).toBe('velopert');
        expect(wrapper.props().name).toBe('ㅈㅊㅎ');

        // dom 검사
        // find에서 css class&id, tag, react component를 인자로 넣을 수 있다.
        const boldElement = wrapper.find('b');
        expect(boldElement.contains('velopert')).toBe(true);
        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('(ㅈㅊㅎ)');
    });
});