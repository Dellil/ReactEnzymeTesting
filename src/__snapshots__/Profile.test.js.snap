// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="장찬희"
  username="aksaksdm"
>
  <div>
    <b>
      aksaksdm
    </b>
     
    <span>
      (
      장찬희
      )
    </span>
  </div>
</Profile>
`;
